src/main/java/org/spongepowered/api/plugin/PluginContainer.java

```diff
@@ -52,10 +57,19 @@
     String getVersion();
 
     /**
-     * Returns the created instance of {@link Plugin}.
+     * Returns the assigned logger to this {@link Plugin}.
+     *
+     * @return The assigned logger
+     */
+    default Logger getLogger() {
+        return LoggerFactory.getLogger(getId());
+    }
+
+    /**
+     * Returns the created instance of {@link Plugin} if it is available.
      *
-     * @return The instance
+     * @return The instance if available
      */
-    Object getInstance();
+    Optional<Object> getInstance();
```

Comment:
Keep in mind we also represent, e.g. Forge mod containers as plugins. Basically in this case there are two options, because some of the Forge containers don't have an instance assigned (currently the method returns null in some cases!):

1. Return a dummy instance or something like `Optional.empty()` as object.
2. Change the method to use an `Optional`